dataflux_pytorch/benchmark/checkpointing/simulated/multi_node_benchmark.py

```diff
@@ -2,52 +2,17 @@
 import socket
 import statistics
 import time
-from typing import Optional, Dict
 
 import torch
-import torch.distributed as dist
-import torch.distributed.checkpoint as dist_cp
-import torch.nn as nn
-from lightning.pytorch.strategies import FSDPStrategy
-from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedWriter, GCSDistributedReader
-
-
-class BenchmarkStrategy(FSDPStrategy):
-    def __init__(self, project: str, path: str, model: nn.Module, **kwargs) -> None:
-        super().__init__(**kwargs)
-        self.writer = GCSDistributedWriter(path, project, None)
-        self.reader = GCSDistributedReader(path, project, None)
-        self.model = model
-
-    def save_checkpoint(self, checkpoint: Dict[str, torch.Tensor], filepath: str, storage_options: Optional[Dict] = None) -> None:
-        dist_cp.save(state_dict=checkpoint, checkpoint_id=filepath,
-                     storage_writer=self.writer)
-
-    def load_checkpoint(self, checkpoint_path: str) -> None:
-        empty_state_dict = {}
-        dist_cp.load(state_dict=empty_state_dict,
-                     checkpoint_id=checkpoint_path, storage_reader=self.reader)
-
-
-class SimpleModel(nn.Module):
-    def __init__(self, size: int, padding_size: int) -> None:
-        super(SimpleModel, self).__init__()
-        self.fc1 = nn.Linear(size, size)
-        self.fc2 = nn.Linear(size, size)
-        self.dummy_tensors = [torch.randn(size, size)
-                              for _ in range(padding_size)]
-
-    def forward(self, x: torch.Tensor) -> torch.Tensor:
-        return self.fc2(torch.relu(self.fc1(x)))
-
-
-def format_size(size_bytes: int) -> str:
-    size_mb = size_bytes / (1024 * 1024)
-    return f"{size_mb / 1024:.2f} GB" if size_mb >= 1024 else f"{size_mb:.2f} MB"
-
-
-def get_tensor_size_bytes(tensor: torch.Tensor) -> int:
-    return tensor.element_size() * tensor.numel()
+from demo.lightning.checkpoint.simulated.multiprocessing_train import (
+    BenchmarkStrategy,
+    SimpleModel,
+    cleanup,
+    format_size,
+    get_tensor_size_bytes,
+    split_tensor,
+    time_checkpoint_operation,
+)
 
 
 def configure_master_addr():
```

Comment:
does this function need to exist if we always expect to run localhost as our host? Is there an example where we don't do that?

Reply:
We do need this in GKE setup, since application runs on multiple pods across different nodes. Each node will have its own IP address.